### PR TITLE
[#1792] Display validation errors for variables

### LINKED ISSUE #1792: Display validation errors for variables
### Thema / Theme

Admin

### Omschrijving / Description

When invalid variables are saved, the backend returns a 400 with the errors. The frontend doesnt display the errors yet.

### Added value / Toegevoegde waarde

_No response_

### Aanvullende opmerkingen / Additional context

_No response_

### PR DESCRIPTION
**Pull request by SilviaAmAm**
Fixes #1792

### CODE REVIEW

**Comment by SilviaAmAm** [comment]
Screenshot from 2022-07-21 16-29-48

File: src/openforms/js/components/admin/form_design/Tab.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import {useIntl} from 'react-intl';
+import {Tab as ReactTab} from 'react-tabs';
+import FAIcon from '../FAIcon';
```
Comment (sergei-maertens) [inline src/openforms/js/components/admin/form_design/Tab.js:3]:
```suggestion
import FAIcon from 'components/admin//FAIcon';
```
& we can group the imports a bit nicer (3rd party / absolute imports / relative imports)

File: src/openforms/js/components/admin/form_design/form-creation-form.js [hunk 2]
```diff
@@ -696,23 +703,43 @@ function reducer(draft, action) {
...
         }
 
+        // Assign errors to variables
+        const variablesValidationErrors = parseValidationErrors(prefixedErrors, 'variables');
+        draft.formVariables = draft.formVariables.map((variable, index) => {
+          return {
+            ...variable,
+            errors: variablesValidationErrors[index],
+          };
+        });
```
Comment (sergei-maertens) [inline src/openforms/js/components/admin/form_design/form-creation-form.js:741]: with immer it's more natural to mutate in place on the draft - something like:

```js
variablesValidationErrors.forEach((errors, index) => {
  draft.formVariables[index].errors = errors;
});
```

File: src/openforms/js/components/admin/form_design/variables/VariablesTable.js [hunk 3]
```diff
@@ -104,84 +106,108 @@ const EditableVariableRow = ({index, variable, onDelete, onChange}) => {
...
-          onBlur={updateKey}
-          noVTextField={true}
-        />
+        <Field name="name" errors={variable.errors?.name}>
+          <TextInput
+            name="name"
+            value={variable.name}
+            onChange={onValueChanged}
+            onBlur={updateKey}
+            noVTextField={true}
```
Comment (sergei-maertens) [inline src/openforms/js/components/admin/form_design/variables/VariablesTable.js:127]:
```suggestion
            noVTextField
```
this is equivalent for boolean props

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

**Review by sergei-maertens — APPROVED** [review]

**Pull request merged**